### PR TITLE
Avatar support

### PR DESCRIPTION
**Pull request by pkostohrys**
Adding gitlab avatar support https://docs.gitlab.com/ee/api/avatar.html

### CODE REVIEW

File: avatar.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+func (s *AvatarRequestsService) GetAvatar(opts *AvatarOptions, options ...RequestOptionFunc) (error, *string) {
+	req, err := s.client.NewRequest(http.MethodGet, "avatar", opts, options)
+	if err != nil {
+		return err, nil
+	}
+
+	avatar := struct {
+		Url string `json:"avatar_url"`
+	}{}
```
Comment (svanharmelen) [inline avatar.go:48]: I prefer to create and return a struct instead:

```
type Avatar struct {
    AvatarURL string `json:"avatar_url"`
}
```

File: avatar.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+// AccessRequest represents a access request for a user avatars.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/avatar.html
+type AvatarOptions struct {
+	Email string `json:"email"`
+	Size  int    `json:"size"`
+}
```
Comment (svanharmelen) [inline avatar.go:30]: These should be pointers, the struct should be places above the `GetAvatar` method and this the struct should be called `GetAvatarOptions`...

File: avatar.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	avatar := struct {
+		Url string `json:"avatar_url"`
+	}{}
+
+	_, err = s.client.Do(req, &avatar)
+	if err != nil {
+		return err, nil
+	}
+
+	return nil, &avatar.Url
```
Comment (svanharmelen) [inline avatar.go:55]: We always return the response as well.

**Review by svanharmelen — CHANGES_REQUESTED** [review]
Thanks @pkostohrys, but it needs some work before we can merge it. Please checkout the rest of the codebase and try to make sure your PR is consistent with it 👍🏻

**Comment by pkostohrys** [comment]
Thank @svanharmelen for you comments. I have pushed an update. Could you please check it now ?

**Review by svanharmelen — APPROVED** [review]
Thanks @pkostohrys! I've updated the code a little to fix the last inconsistencies. Please check the latest commit to see what I've changed 👍🏻

**Pull request merged**